src/Command/MergeMigrationsCommand.php

```diff
@@ -0,0 +1,214 @@
+                InputArgument::REQUIRED,
+                'The target Tag number (dots will be converted to underscores)'
+            )
+            ->setHelp(<<<EOT
+The <info>%command.name%</info> command will merge all migrations generated with 
+doctrine:migrations:diff or doctrine:migrations:generate commands to one corresponding to a specific tag name.
+
+<info>php %command.full_name% [--tag=...]</info>
+
+EOT
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $this->init();
+
+        $io->title('Merge Doctrine Migrations');
+
+        $tag = $input->getArgument('tag');
+
+        $versions = $this->handlingExistingMigrations();
+
+        $this->createTargetMigration($tag, $versions);
+
+        return 0;
+    }
+
+    /**
+     * Init parameters and loads migration classes.
+     */
+    private function init()
+    {
+        $this->doctrineMigrationPath = $this->container->getParameter('doctrine_migrations.dir_name');
+        $this->doctrineMigrationNamespace = $this->container->getParameter('doctrine_migrations.namespace');
+
+        $finder = new Finder();
+        $finder->files()->in($this->doctrineMigrationPath)->name('/^Version[0-9]{14}\.php/');
+
+        // Loading migrations classes
+
+        /** @var SplFileInfo $file */
+        foreach ($finder as $file) {
+            include $file->getRealPath();
+        }
+    }
+
+    /**
+     * Undocumented function.
+     *
+     * @return array
+     */
+    private function handlingExistingMigrations(): array
+    {
+        // Retrieves migrations from namespaces
+
+        $doctrineMigrationNamespace = $this->doctrineMigrationNamespace;
+        $migrations = array_filter(get_declared_classes(), function ($class) use ($doctrineMigrationNamespace) {
+            preg_match('/^' . $doctrineMigrationNamespace . '\\\/', $class, $found);
+
+            return count($found) > 0;
+        });
+
+        // Looping through versions to build methods definitions
+
+        $versions = [];
+
+        foreach ($migrations as $migrationClass) {
+            $reflectionClass = new \ReflectionClass($migrationClass);
+
+            $versionCode = str_replace('Version', '', $reflectionClass->getShortName());
+
+            $fn = [
+                'up'   => null,
+                'down' => null,
+            ];
+
+            // Handling up method
+
+            $up = $reflectionClass->getMethod('up');
+
+            $fn['up'] = [
+                'filePath'  => $reflectionClass->getFileName(),
+                'startLine' => $up->getStartLine(),
+                'endLine'   => $up->getEndLine(),
+                'body'      => '',
+            ];
+
+            $fn['up']['body'] = implode('', array_slice(file($fn['up']['filePath']), $fn['up']['startLine'], ($fn['up']['endLine'] - $fn['up']['startLine']) + 1));
```

Comment:
line break ?

Reply:
Oups, the lien break does nothing but I've missed to replace this part by calling `$this->getMethodInfos($up)`